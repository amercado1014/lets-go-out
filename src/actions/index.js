export const addRestaurants = restaurants => ({
  type: 'ADD_RESTAURANTS',
  restaurants
});

export const addMenu = menu => ({
  type: 'ADD_MENU',
  menu
});

export const addRestaurant = restaurant => ({
  type: "ADD_RESTAURANT",
  restaurant
});

export const locationOff = boolean => ({
  type: 'LOCATION_OFF',
  boolean
});

export const addUser = user => ({
  type: 'ADD_USER',
  user
});

export const addFavorite = favorite => ({
  type: 'ADD_FAVORITE',
  favorite
});

export const removeFavorite = apiKey => ({
  type: 'REMOVE_FAVORITE',
  apiKey
});

export const resetFavorites = () => ({
  type: 'RESET_FAVORITES'
});