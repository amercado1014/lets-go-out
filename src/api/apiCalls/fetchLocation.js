import { apiKey2 } from "./apiKey";

export const fetchLocation = async () => {
  const root = "https://www.googleapis.com/geolocation/v1";
  const url = `${root}/geolocate?key=${apiKey2}`;
  try {
    const response = await fetch(url, {
      method: "POST",
      headers: { "Content-Type": "application/json" }
    });
    const location = await response.json();
    return location.location;
  } catch (error) {
    throw new Error(`Error fetching location: ${error.message}`);
  }
};
