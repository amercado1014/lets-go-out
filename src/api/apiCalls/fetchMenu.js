import { apiKey } from "./apiKey";

export const fetchMenu = async menuKey => {
  const root = "https://api.eatstreet.com/publicapi/v1";
  const url = `${root}/restaurant/${menuKey}/menu?includeCustomizations=false`;
  try {
    const response = await fetch(url, {
      method: "GET",
      headers: { "X-Access-Token": apiKey }
    });
    const menu = await response.json();
    return menu;
  } catch (error) {
    throw new Error("Error fetching menu: ", error);
  }
};
