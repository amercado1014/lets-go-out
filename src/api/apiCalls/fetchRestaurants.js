import apiKey from './apiKey';

export const fetchRestaurants = async search => {
  const root = 'https://api.eatstreet.com/publicapi/v1'; 
  const url = `${root}/restaurant/search?method=both&street-address=${search}`;
  try {
    const response = await fetch(url, {
      method: "GET",
      headers: { "X-Access-Token": apiKey }
    });
    const restaurants = await response.json();
    return restaurants.restaurants;
  } catch (error) {
    throw new Error('Error fetching restaurants: ', error);
  }
};
