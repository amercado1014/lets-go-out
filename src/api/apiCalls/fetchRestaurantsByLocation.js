import { apiKey } from './apiKey';

export const fetchRestaurantsByLocation = async location => {
  const root = 'https://api.eatstreet.com/publicapi/v1';
  // eslint-disable-next-line 
  const url = `${root}/restaurant/search?latitude=${location.lat}&longitude=${location.lng}&method=both`;
  try {
    const response = await fetch(url, {
      method: "GET",
      headers: { "X-Access-Token": apiKey }
    });
    const restaurants = await response.json();
    return restaurants.restaurants;
  } catch (error) {
    throw new Error(`Error fetching restaurants: ${error.details}`);
  }
};
