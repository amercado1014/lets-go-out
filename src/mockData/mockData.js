export const mockRestaurantsApiData = {
  "address": {
    "apiKey": null,
    "streetAddress": "",
    "latitude": 39.6682515,
    "longitude": -105.096661,
    "city": "Denver",
    "state": "CO",
    "zip": "80227",
    "aptNumber": null
  },
  "restaurants": [
    {
      "apiKey": "aa1163f8eb68a8f7144247b91fbcb2f31932e981e6050b6c",
      "deliveryMin": 0,
      "deliveryPrice": 3.5,
      "logoUrl": "https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-s-wadsworth-21136_1407870747356.png",
      "name": "Big Daddy's Pizza - S. Wadsworth",
      "streetAddress": "2772 S Wadsworth Blvd",
      "city": "Denver",
      "state": "CO",
      "zip": "80227",
      "foodTypes": [
        "Pizza"
      ],
      "phone": "(303) 988-3697",
      "latitude": 39.666141,
      "longitude": -105.080677,
      "minFreeDelivery": 0,
      "taxRate": 0.08,
      "acceptsCash": true,
      "acceptsCard": true,
      "offersPickup": true,
      "offersDelivery": true,
      "isTestRestaurant": false,
      "minWaitTime": 45,
      "maxWaitTime": 60,
      "open": false,
      "url": "https://eatstreet.com/denver-co/restaurants/big-daddys-pizza-s-wadsworth",
      "hours": {
        "Monday": [
          "11:00 AM - 3:00 AM"
        ],
        "Saturday": [
          "11:00 AM - 3:00 AM"
        ],
        "Sunday": [
          "11:00 AM - 3:00 AM"
        ],
        "Wednesday": [
          "11:00 AM - 3:00 AM"
        ],
        "Tuesday": [
          "11:00 AM - 3:00 AM"
        ],
        "Friday": [
          "11:00 AM - 3:00 AM"
        ],
        "Thursday": [
          "11:00 AM - 3:00 AM"
        ]
      },
      "timezone": "US/Mountain"
    }
  ]
};

export const returnedMockRestaurantApiData = [
  {
    "apiKey": "aa1163f8eb68a8f7144247b91fbcb2f31932e981e6050b6c",
    "deliveryMin": 0,
    "deliveryPrice": 3.5,
    "logoUrl": "https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-s-wadsworth-21136_1407870747356.png",
    "name": "Big Daddy's Pizza - S. Wadsworth",
    "streetAddress": "2772 S Wadsworth Blvd",
    "city": "Denver",
    "state": "CO",
    "zip": "80227",
    "foodTypes": [
      "Pizza"
    ],
    "phone": "(303) 988-3697",
    "latitude": 39.666141,
    "longitude": -105.080677,
    "minFreeDelivery": 0,
    "taxRate": 0.08,
    "acceptsCash": true,
    "acceptsCard": true,
    "offersPickup": true,
    "offersDelivery": true,
    "isTestRestaurant": false,
    "minWaitTime": 45,
    "maxWaitTime": 60,
    "open": false,
    "url": "https://eatstreet.com/denver-co/restaurants/big-daddys-pizza-s-wadsworth",
    "hours": {
      "Monday": [
        "11:00 AM - 3:00 AM"
      ],
      "Saturday": [
        "11:00 AM - 3:00 AM"
      ],
      "Sunday": [
        "11:00 AM - 3:00 AM"
      ],
      "Wednesday": [
        "11:00 AM - 3:00 AM"
      ],
      "Tuesday": [
        "11:00 AM - 3:00 AM"
      ],
      "Friday": [
        "11:00 AM - 3:00 AM"
      ],
      "Thursday": [
        "11:00 AM - 3:00 AM"
      ]
    },
    "timezone": "US/Mountain"
  }
];

export const cleanRestaurantData = {
  apiKey: "aa1163f8eb68a8f7144247b91fbcb2f39e65e39821ec08a9",
  logoUrl: "https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-lakewood-21135_1407868711096.png",
  name: "Big Daddy's Pizza - Lakewood",
  streetAddress: "850 Wadsworth Blvd",
  city: "Lakewood",
  state: "CO",
  zip: "80214",
  foodTypes: "Pizza",
  phone: "(303) 736-6090",
  url: "https://eatstreet.com/lakewood-co/restaurants/big-daddys-pizza-lakewood",
  hours: {
    Monday: "11:00 AM - 3:00 AM",
    Saturday: "11:00 AM - 3:00 AM",
    Sunday: "11:00 AM - 3:00 AM",
    Wednesday: "11:00 AM - 3:00 AM",
    Tuesday: "11:00 AM - 3:00 AM",
    Friday: "11:00 AM - 3:00 AM",
    Thursday: "11:00 AM - 3:00 AM"
  }
};