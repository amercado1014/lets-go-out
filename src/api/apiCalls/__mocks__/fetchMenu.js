import { mockMenu } from '../../../mockData/mockData';

export const fetchMenu = jest.fn().mockImplementation(() => {
  return Promise.resolve({ mockMenu });
});