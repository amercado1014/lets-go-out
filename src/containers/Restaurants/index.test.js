import React from 'react';
import { Restaurants,
  mapStateToProps, 
  mapDispatchToProps } from '../Restaurants';
import { shallow } from 'enzyme';
import { cleanRestaurantData, mockMenu } from '../../mockData/mockData';
import { fetchMenu } from '../../api/apiCalls/fetchMenu';
import * as actions from '../../actions';

jest.mock('../../api/apiCalls/fetchMenu');

describe('Restaurants', () => {
  let wrapper;
  let mockAddFavorite;
  let mockRemoveFavorite;
  const mockAddMenu = jest.fn();
  const mockAddRestaurant = jest.fn();
  const mockHistory = { push: jest.fn() };

  beforeEach(() => {
    mockAddFavorite = jest.fn();
    mockRemoveFavorite = jest.fn();
    wrapper = shallow(<Restaurants restaurant={cleanRestaurantData} 
      addMenu={mockAddMenu}
      addRestaurant={mockAddRestaurant}
      addFavorite={mockAddFavorite}
      removeFavorite={mockRemoveFavorite} 
      history={mockHistory}/>);
  });

  it('should math the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an empty string as default state for error', () => {
    expect(wrapper.state('error')).toEqual('');
  });

  it('should call fetchMenu on handleClick with correct params', () => {
    wrapper.instance().handleClick();
    expect(fetchMenu).toHaveBeenCalledWith(cleanRestaurantData.apiKey);
  });

  it('should call addMenu on handleClick with correct params', () => {
    wrapper.instance().handleClick();
    expect(mockAddMenu).toHaveBeenCalledWith({mockMenu});
  });

  it('should call addRestaurant on handleClick with correct params', () => {
    wrapper.instance().handleClick();
    expect(mockAddRestaurant).toHaveBeenCalledWith(cleanRestaurantData);
  });

  it('should call history.push with correct params on handleClick', () => {
    wrapper.instance().handleClick();
    expect(mockHistory.push).toHaveBeenCalledWith('/menu');
  });

  it('should call addFavorite on addFavorite with correct params', () => {
    wrapper.instance().addFavorite();
    const expected = cleanRestaurantData;
    expect(mockAddFavorite).toHaveBeenCalledWith(expected);
  });

  it('should call removeFavorite on removeFavorite with correct params', () => {
    const wrapper = shallow(<Restaurants restaurant={cleanRestaurantData} 
      removeFavorite={mockRemoveFavorite} 
      isFavorite={true}/>);
    wrapper.instance().removeFavorite();
    const expected = cleanRestaurantData.apiKey;
    expect(mockRemoveFavorite).toHaveBeenCalledWith(expected);
  });

  it('should call addFavorite on handleFavorite', () => {
    wrapper.instance().addFavorite();
    expect(mockAddFavorite).toHaveBeenCalled();
  });

  it('should call removeFavorite on handleFavorite', () => {
    const wrapper = shallow(<Restaurants restaurant={cleanRestaurantData} 
      removeFavorite={mockRemoveFavorite} 
      isFavorite={true} />);
    wrapper.instance().removeFavorite();
    expect(mockRemoveFavorite).toHaveBeenCalled();
  });
});

describe("mapStateToProps", () => {
  it("correctly maps authUser to props", () => {
    const authUser = { name: "Jimmy" };
    const expected = authUser;
    const mockState = { authUser };
    const mapped = mapStateToProps(mockState);
    expect(mapped.authUser).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with correct params on addMenu', () => {
    const mockDispatch = jest.fn();
    const menu = mockMenu;
    const expected = actions.addMenu(menu);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addMenu(menu);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch with correct params on addRestaurant', () => {
    const mockDispatch = jest.fn();
    const restaurant = cleanRestaurantData;
    const expected = actions.addRestaurant(restaurant);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addRestaurant(restaurant);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch with correct params on addFavorite', () => {
    const mockDispatch = jest.fn();
    const restaurant = cleanRestaurantData;
    const expected = actions.addFavorite(restaurant);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addFavorite(restaurant);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it('should call dispatch with the correct params on removeFavorite', () => {
    const mockDispatch = jest.fn();
    const apiKey = cleanRestaurantData.apiKey;
    const expected = actions.removeFavorite(apiKey);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.removeFavorite(apiKey);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});
