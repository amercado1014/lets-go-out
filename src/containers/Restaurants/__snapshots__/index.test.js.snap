// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurants should math the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Restaurants
    restaurant={
        Object {
            "apiKey": "aa1163f8eb68a8f7144247b91fbcb2f39e65e39821ec08a9",
            "city": "Lakewood",
            "deliveryMin": 12,
            "deliveryPrice": 3.5,
            "foodTypes": "Pizza",
            "hours": Object {
              "Friday": "11:00 AM - 3:00 AM",
              "Monday": "11:00 AM - 3:00 AM",
              "Saturday": "11:00 AM - 3:00 AM",
              "Sunday": "11:00 AM - 3:00 AM",
              "Thursday": "11:00 AM - 3:00 AM",
              "Tuesday": "11:00 AM - 3:00 AM",
              "Wednesday": "11:00 AM - 3:00 AM",
            },
            "logoUrl": "https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-lakewood-21135_1407868711096.png",
            "minFreeDelivery": 0,
            "name": "Big Daddy's Pizza - Lakewood",
            "offersDelivery": true,
            "phone": "(303) 736-6090",
            "state": "CO",
            "streetAddress": "850 Wadsworth Blvd",
            "url": "https://eatstreet.com/lakewood-co/restaurants/big-daddys-pizza-lakewood",
            "zip": "80214",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Big Daddy's Pizza - Lakewood
</h2>,
        <img
          alt="restaurant logo"
          src="https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-lakewood-21135_1407868711096.png"
/>,
        <p>
          Minimum: 
          12
</p>,
        <p>
          Delivery: 
          3.5
</p>,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Big Daddy's Pizza - Lakewood",
        },
        "ref": null,
        "rendered": "Big Daddy's Pizza - Lakewood",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "restaurant logo",
          "src": "https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-lakewood-21135_1407868711096.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Minimum: ",
            12,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Minimum: ",
          12,
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Delivery: ",
            3.5,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Delivery: ",
          3.5,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Big Daddy's Pizza - Lakewood
</h2>,
          <img
            alt="restaurant logo"
            src="https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-lakewood-21135_1407868711096.png"
/>,
          <p>
            Minimum: 
            12
</p>,
          <p>
            Delivery: 
            3.5
</p>,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Big Daddy's Pizza - Lakewood",
          },
          "ref": null,
          "rendered": "Big Daddy's Pizza - Lakewood",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "restaurant logo",
            "src": "https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-lakewood-21135_1407868711096.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Minimum: ",
              12,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Minimum: ",
            12,
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Delivery: ",
              3.5,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Delivery: ",
            3.5,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
